### PR TITLE
[examples] Match v5 compatibility

### PR DESCRIPTION
**Pull request by bharatkashyap**
- [x] Update example files to match v5 compatibility updates
- [x] Change `AUTH_GITHUB_ID` to `GITHUB_CLIENT_ID` in examples to match `create-toolpad-app` behaviour
- [x] Fix a hydration error when using Pages Router on `SignInPage`

### CODE REVIEW

**Review by Janpot — APPROVED** [review]

File: packages/toolpad-core/src/SignInPage/SignInPage.tsx [hunk 1]
```diff
@@ -141,6 +141,12 @@ function SignInPage(props: SignInPageProps) {
       ? (new URLSearchParams(window.location.search).get('callbackUrl') ?? '/')
       : '/';
 
+  const [isClient, setIsClient] = React.useState(false);
```
Comment (Janpot) [inline packages/toolpad-core/src/SignInPage/SignInPage.tsx:144]: The context should be there serverside.

File: packages/toolpad-core/package.json [hunk 2]
```diff
@@ -39,7 +40,8 @@
   "scripts": {
     "clean": "rimraf build",
     "prebuild": "pnpm clean",
-    "build": "pnpm build:node && pnpm build:stable && pnpm build:types && pnpm build:copy-files",
+    "build": "pnpm build:node && pnpm build:stable && pnpm build:types && pnpm build:copy-files && pnpm esmify",
+    "esmify": "rm -rf ./build/node && rm -rf ./build/*/package.json",
```
Comment (Janpot) [inline packages/toolpad-core/package.json:44]: Next.js resolver gets very confused about these `@toolpad/core/*/package.json` files. It loads CJS from `pages/_app.tsx` and ESM from `pages/auth/signin.tsx`. Forcing the package to be ESM-only solves all issues of double contexts being created.

Will adapt the build script in core with a mode for esm only.

File: packages/toolpad-core/src/SignInPage/SignInPage.tsx [hunk 3]
```diff
@@ -136,10 +136,9 @@ function SignInPage(props: SignInPageProps) {
     error: '',
   });
 
-  const callbackUrl =
-    typeof window !== 'undefined'
-      ? (new URLSearchParams(window.location.search).get('callbackUrl') ?? '/')
-      : '/';
+  const router = React.useContext(RouterContext);
```
Comment (Janpot) [inline packages/toolpad-core/src/SignInPage/SignInPage.tsx:139]: cc @bharatkashyap

**Review by Janpot — APPROVED** [review]

File: examples/core-auth-nextjs-pages/src/pages/auth/signin.tsx [hunk 4]
```diff
@@ -49,6 +44,7 @@ export default function SignIn({
           return {};
         } catch (error) {
           // An error boundary must exist to handle unknown errors
+          console.log('Is this happening');
```
Comment (apedroferreira) [inline examples/core-auth-nextjs-pages/src/pages/auth/signin.tsx:47]: I think you missed this console log, just saw it :D

Comment (apedroferreira) [reply examples/core-auth-nextjs-pages/src/pages/auth/signin.tsx:47]: I can remove it in the release PR.

Comment (bharatkashyap) [reply examples/core-auth-nextjs-pages/src/pages/auth/signin.tsx:47]: Missed that, thanks for that @apedroferreira :)

**Pull request merged**